src/plugins/intel_gna/tests/functional/preprocess_tests/gather_transpose_cpu.cpp

```diff
@@ -0,0 +1,234 @@
+        auto matmul_node = std::make_shared<Multiply>(pre_node, mul_input_const);
+
+        auto add_input_const = Constant::create(m_net_type,
+                                                pre_node->get_shape(),
+                                                CommonTestUtils::generate_float_numbers(input_shape_size, -0.2f, 0.2f));
+        auto add_node = std::make_shared<Add>(matmul_node, add_input_const);
+
+        std::shared_ptr<ov::Node> post_node = add_node;
+        switch (m_post_layer) {
+        case PrePostLayer::GATHER:
+            pre_node = make_gather_node(post_node);
+            break;
+        case PrePostLayer::TRANSPOSE:
+            pre_node = make_transpose_node(post_node);
+            break;
+        default:
+            break;
+        }
+
+        ov::ResultVector results{std::make_shared<Result>(post_node)};
+        function = std::make_shared<ov::Model>(results, params, "concat");
+    }
+
+    std::shared_ptr<ov::Node> make_gather_node(std::shared_ptr<ov::Node> node) {
+        const std::vector<size_t> indexes = make_indexes(ov::shape_size(m_input_shape));
+        auto gather_indexes = Constant::create(ov::element::i64, ov::Shape{indexes.size()}, indexes);
+        const size_t gather_axis = m_input_shape.size() - 1;
+        auto gather_axis_const = Constant::create(ov::element::i64, ov::Shape{}, {gather_axis});
+        auto gather_node = std::make_shared<Gather>(node, gather_indexes, gather_axis_const);
+
+        return gather_node;
+    }
+
+    std::shared_ptr<ov::Node> make_transpose_node(std::shared_ptr<ov::Node> node) {
+        std::vector<size_t> transpose_order = make_transpose_order(m_input_shape);
+        ov::Shape transpose_shape;
+        for (size_t i : transpose_order) {
+            transpose_shape.push_back(m_input_shape[i]);
+        }
+
+        auto transpose_const =
+            std::make_shared<Constant>(ov::element::i8, ov::Shape{transpose_order.size()}, transpose_order);
+        auto transpose_node = std::make_shared<Transpose>(node, transpose_const);
+
+        return transpose_node;
+    }
+
+    ov::element::Type m_net_type;
+    std::vector<size_t> m_input_shape;
+    PrePostLayer m_pre_layer = PrePostLayer::NONE;
+    PrePostLayer m_post_layer = PrePostLayer::NONE;
+};
+
+TEST_P(PrePostProcessBaseTest, CompareWithRefs) {
+    init_test_model();
+    run();
+}
+
+std::vector<std::map<std::string, std::string>> configs = {{{"GNA_DEVICE_MODE", "GNA_SW_EXACT"}}};
+
+std::vector<std::map<std::string, std::string>> target_configs = {{{"GNA_DEVICE_MODE", "GNA_SW_FP32"}},
+                                                                  {{"GNA_EXEC_TARGET", "GNA_TARGET_2_0"}},
+                                                                  {{"GNA_EXEC_TARGET", "GNA_TARGET_3_0"}},
+                                                                  {{"GNA_EXEC_TARGET", "GNA_TARGET_3_5"}}};
```

Comment:
are you sure you want to add this now?
{{"GNA_EXEC_TARGET", "GNA_TARGET_3_5"}}};

Reply:
why not?